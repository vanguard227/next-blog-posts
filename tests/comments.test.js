import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import Comments from "../components/comments";

describe("<Comments />", () => {
  const comments = [
    {
      id: "1",
      title: "Comment1",
      description: "Comment description",
      createdAt: "2020-10-07T21:01:51.705Z",
    },
  ];

  it("matches snapshot", () => {
    const component = renderer.create(
      <Comments
        comments={comments}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders comments", () => {
    const wrapper = shallow(<Comments comments={comments} />);
    let actual = wrapper.find("h6").text();
    expect(actual).toBe(comments[0].title);

    actual = wrapper.find("span").text();
    const expected = new Date(comments[0].createdAt).toLocaleDateString();
    expect(actual).toBe(expected);

    actual = wrapper.find("p").text();
    expect(actual).toBe(comments[0].description);

    actual = wrapper.find("div").length;
    expect(actual).toBe(comments.length);
  });
});
