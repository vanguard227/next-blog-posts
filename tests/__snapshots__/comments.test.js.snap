// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comments /> matches snapshot 1`] = `
<div
  className="sc-bczRLJ feWHbK"
>
  <div>
    <h6>
      Comment1
    </h6>
    <span>
      10/7/2020
    </span>
    <p>
      Comment description
    </p>
  </div>
</div>
`;
